source/_integrations/huawei_lte.markdown

```diff
@@ -1,161 +1,122 @@
+authentication to work varies by device and firmware version. The
+integration will try to use all configured ones and fail gracefully if
+it detects one requiring authentication in unauthenticated mode.
 
-{% configuration %}
-url:
-  description: URL of the router web interface.
-  required: true
-  type: string
-username:
-  description: The username used for the router web interface.
-  required: true
-  type: string
-password:
-  description: The password used for the router web interface.
-  required: true
-  type: string
-{% endconfiguration %}
-
-### Tested routers
+Only a subset of the entities provided by the target device by
+default: 
 
-Routers we know to be working with this integration based on the documentation of used libraries and reports by users:
+- WAN IP address sensor
+- RSRQ sensor
+- RSRP sensor
+- RSSI sensor
+- SINR LTE signal sensor
+- mobile data switch
+- device tracker entries
 
-- Huawei B310s-22
-- Huawei B525s-23a
-- Huawei E5186s-22a
-- Huawei B618
+The rest are added to the entity registry, but disabled by default.
 
-This is not a complete list. The integration can probably connect to other Huawei LTE routers running similar firmware.
+Support for different categories of information and thus available
+entities varies by device model and firmware version.
 
-## Presence Detection
+### Configuration via the frontend
 
-This platform offers presence detection by looking at connected devices to a [Huawei LTE router](https://consumer.huawei.com/en/smart-home/).
+Menu: **Configuration** -> **Integrations**.
 
-To enable the sensor, add the following lines to your `configuration.yaml` file:
+Click on the `+` sign to add an integration and click on **Huawei
+LTE**, and follow the configuration flow. After finishing, the Huawei
+LTE integration will be available.
 
-```yaml
-# Example configuration.yaml entry
-device_tracker:
-  - platform: huawei_lte
-```
+To use unauthenticated mode, leave username and password empty. The
+integration will then attempt to first use empty strings to
+authenticate and fall back to unauthenticated mode in case that
+fails. If this process does not yield desired results, the YAML
+configuration (see below) is available for more fine grained control.
 
-See the [device tracker integration page](/integrations/device_tracker/) for instructions how to configure the people to be tracked.
+Default list of notification recipient phone numbers can be set using
+the integration's configuration options.
 
-## Notifications
+### Configuration via YAML
 
-The `huawei_lte` platform allows you to use a Huawei LTE router for notifications from Home Assistant. The messages will be sent as SMS text messages.
+To enable the component, add the following lines to your
+`configuration.yaml` file:
 
 ```yaml
 # Example configuration.yaml entry
-notify:
-  - platform: huawei_lte
-    recipient: "+15105550123"
+huawei_lte:
+  - url: http://192.168.100.1/
+    username: YOUR_USERNAME
```

Comment:
We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

Reply:
While username and password are really optional, in practice I believe the vast majority of people are going to be using authenticated mode, i.e. with username + password. Without them, in unauthenticated mode, many useful integration features are disabled.

So I left username and password there for now, but I don't feel that strongly about this. If you still think they should be removed, please feel free to go ahead and remove or let me know and I'll take care of it.

Reply:
I see your point of view, but in general, we follow these rules to get a more consistent experience in the documentation. IMHO, it makes no sense to accept it in one place and deny it in another.

Reply:
Ok, removed.